pkg/resourcemanager/controller/add.go

```diff
@@ -43,6 +44,16 @@ func AddToManager(mgr manager.Manager, sourceCluster, targetCluster cluster.Clus
 		return fmt.Errorf("failed creating Kubernetes client: %w", err)
 	}
 
+	targetServerVersion, err := targetClientSet.Discovery().ServerVersion()
+	if err != nil {
+		return err
+	}
+
+	targetKubernetesVersion, err := semver.NewVersion(targetServerVersion.GitVersion)
+	if err != nil {
+		return err
+	}
```

Comment:
What is the benefit of parsing the semver here if you use `versionutils.CheckVersionMeetsConstraint` with the string representation later on?